interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ExpressionCompiler.scala

```diff
@@ -219,9 +228,12 @@ class ExpressionCompiler(expressionParsers: Map[String, ExpressionParser]) {
   }
 
   def withExpressionParsers(modify: PartialFunction[ExpressionParser, ExpressionParser]): ExpressionCompiler =
-    new ExpressionCompiler(expressionParsers.map { case (k, v) =>
-      k -> modify.lift(v).getOrElse(v)
-    })
+    new ExpressionCompiler(
+      expressionParsers.map { case (k, v) =>
+        k -> modify.lift(v).getOrElse(v)
+      },
+      dictRegistry
```

Comment:
Take a look for the usage of this method :) There is only one:
```
uiValidator.withValidator(_.withExpressionParsers {
    case spel: SpelExpressionParser =>
      spel.typingDictLabels
  })
```
The problem that we have is that we don't want to query labels on the engine side and this problem I think is still valid for your case - I'm not sure about that. Even if this method works correctly for your case, I think that we need to change the abstraction level of this method to avoid confusions e.g. rename it to `typingDictLabels`, rename `dictRegistry` to `dcitParameterRegistry` and add a comment that during dict parameter validation, there is always a key inside the expression.

Reply:
renamed/clarified it into `withLabelsDictTyper`, as that was it's only usage